libs/linAlg/linAlg.cpp

```diff
@@ -34,132 +34,280 @@ namespace libp {
 /*********************/
 
 // o_a[n] = alpha
-void linAlg_t::set(const dlong N, const dfloat alpha, deviceMemory<dfloat> o_a) {
-  setKernel(N, alpha, o_a);
+template<>
+void linAlg_t::set(const dlong N, const float alpha, deviceMemory<float> o_a) {
+  setKernelFloat(N, alpha, o_a);
+}
+
+template<>
+void linAlg_t::set(const dlong N, const double alpha, deviceMemory<double> o_a) {
+  setKernelDouble(N, alpha, o_a);
```

Comment:
Lots of repeated code here. Maybe better to have one templated implementation and just instantiate it for float and double afterwards.